app/src/Slider.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import { render } from "react-dom";
+import { Slider, Rail, Handles, Tracks, Ticks } from "react-compound-slider";
+import { SliderRail, Handle, Track, Tick } from "./components"; // example render components - source below
+import { format} from "date-fns";
+import { scaleTime } from "d3-scale";
+
+const sliderStyle = {
+  position: "relative",
+  width: "100%"
+};
+
+function formatTick(ms) {
+  return format(new Date(ms), "yyyy");
+}
+
+const oneYear = 1000 * 60 * 60 * 24 * 365;
+
+// Time
+class Slider2 extends Component {
+	constructor() {
+	    super();
+
+	   	const defaultYear = new Date(1950, 3, 3);
+	    const startYear = new Date(1950, 3, 3);
+	    const endYear = new Date(2019, 3, 3);
+
+	    this.state = {
+	      selected: defaultYear,
+	      updated: defaultYear,
+	      min: startYear,
+	      max: endYear
+	    };
+	  }
+
+	  onChange = ([ms]) => {
```

Comment:
I think you have to bind this in the constructor --> `this.onChange = this.onChange.bind(this)` see --> https://reactjs.org/docs/handling-events.html